tests/sitenotice_test.php

```diff
@@ -78,6 +78,30 @@ private function create_cohort_notice2() {
         helper::create_new_notice($formdata);
     }
 
+    /**
+     * Create sample notice (with a start and end date).
+     */
+    private function create_notice_with_start_and_end1() {
+        $formdata = new \stdClass();
+        $formdata->title = "Date range Notice 1";
+        $formdata->content = "Date range Notice 1 <a href=\"www.example3.com\">Link 3</a> <a href=\"www.example4.com\">Link 9</a>";
+        $formdata->timestart = time() + HOURSECS;
+        $formdata->timeend = time() + DAYSECS;
+        helper::create_new_notice($formdata);
+    }
+
+    /**
+     * Create sample notice (with a start and end date).
+     */
+    private function create_notice_with_start_and_end2() {
```

Comment:
I don't think having methods like blabla1 and blabla2 is the best practice. Can you please rewrite to have one generic method where you can pass all required data from your test?

I see that the code already does that. If you can fix existing code that would be a bonus.